aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,740 @@
+ 
+Expression encoding
+-------------------
+
+Non-:dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent` content is HTML encoded. For example, the following Razor markup:
+
+.. code-block:: none
+
+  @("<span>Hello World</span>") 
+
+Renders this HTML:
+
+.. code-block:: none
+
+  &lt;span&gt;Hello World&lt;/span&gt;
+  
+Which the browser renders as:
+
+``<span>Hello World</span>`` 
+
+:dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` :dn:method:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.Raw` output is not encoded but rendered as HTML markup. :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` implements :dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent`
+
+.. warning:: Using ``HtmlHelper.Raw`` on unsanitzed user input is a security risk. User input might contain malicious JavaScript or other exploits. Sanitizing user input is difficult, avoid using ``HtmlHelper.Raw`` on user input.
+
+The following Razor markup:
+
+.. code-block:: none
+
+  @Html.Raw("<span>Hello World</span>") 
+
+Renders this HTML:
+
+.. code-block:: none
+
+  <span>Hello World</span> 
+  
+Which the browser renders as:
+``Hello World`` 
+
+Rendering markup in code blocks and implicit transition
+-------------------------------------------------------
+
+The default languge in a code block is c#, but you can transition back to HTML. HTML within a code block will transition back into rendering HTML:
+
+.. code-block:: none
+
+  @{
+      var inCSharp = true;
+      <p>Now in HTML, was in C# @inCSharp</p>
+  }
+
+
+Explicit delimited transition
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To define a sub-section of a code block that should render HTML, surround the characters to be rendered with the ``<text>`` tag:
+
+.. code-block:: none
+
+  @{
+  /* C# */<text>I'm HTML</text>/* C# */
+  }
+
+Which renders ``I'm HTML`` without the ``text`` tags. You generally use this approach when you want to render HTML that is not surrounded by an HTML tag.
+
+.. _Explicit-Line-Transition-label:
+
+Explicit Line Transition with ``@:``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To render an entire line inside of a code block, utilize the ``@:`` characters:
+
+.. code-block:: none
+
+  @{
+  /* Still C# */@: <p>Hello World</p> /* This is not C#, it's HTML */
+  }
+
+Which renders the following HTML:
+
+.. code-block:: none
+
+  <p>Hello World</p> /* This is not C#, it's HTML */
+  
+And is rendered in a browser as:
+ 
+.. code-block:: none
+
+  Hello World
+
+  /* This is not C#, it's HTML */ 
+  
+Consider the following Razor markup which renders a list of names:
+
+.. code-block:: none
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      <p>Name: @person.Name</p>
+  }
+
+HTML tags (``<p> </p>`` in the sample, but any HTML tags will do) provide a boundry for Razor to transition into C#. But suppose you wanted to render the names **without** HTML tags? The following code generates a Razor compilation error:
```

Comment:
Personally, I still prefer the minimalistic approach. There's tons of stuff that we could show people how to do in Razor, but it doesn't all fit in this doc. This is just supposed to be a Razor syntax reference. There will be plenty of opportunities to show people how to do things on top of this basic knowledge (blog posts, tutorials, SO, etc.)